### DISCUSSION TITLE
Make `create-next-app` work with private examples

### DISCUSSION THREAD

**Discussion by Jared-Dev** [vercel/next.js#19218] [Ideas]
# Feature request
Please provide a way to use a private repo as the source when using `create-next-app`
## Is your feature request related to a problem? Please describe.
Currently we are using other tools to clone/update references for fresh projects. It would be great if we could use the tools that Next provides!

## Describe the solution you'd like
Currently when using an example other than default with `create-next-app` it is using `http`. It would be great if we could pass in a PAT for private repos. Or if there was a way to have the script use the native `git` client that might be a great alternative!

## Describe alternatives you've considered
Currently we are spinning up a Laravel project alongside our Next.JS frontend so we have had to hack in some workarounds since private repos don't work.

**Comment by mryechkin** [comment]
Is this still not doable? I'm trying to use `create-next-app` with a private GitHub repo and getting an error: "Found invalid GitHub URL". 

The URL I'm specifying is of the format `https://github.com/{organization}/{repo-name}`

**Comment by gwenshap** [comment]
It is fairly straightforward to add a `--example-token` where the developer can provide github access token that allows access to private repos, and then use this token as auth bearer header when calling Github (per Github's API docs). 

Is this solution acceptable? I can contribute a PR in that case (although it looks like NextJS is already dealing with quite a PR backlog!).
